### DISCUSSION TITLE
Ceph dashboard Audit Logs

### DISCUSSION THREAD

**Discussion by serenagrl** [rook/rook#11052] [Q&A]
Hi, 

I am new to rook-ceph and I would like to know if there is a way to set the Audit Log level that is shown in the Ceph dashboard -> Logs -> Audit Log tab. In a fresh install, it is currently creating 4 records every minute which looks something like the following:
```
9/26/22 12:29:49 PM
[INF]
from='mgr.54140 ' entity='mgr.b' cmd=[{"prefix":"config rm","who":"mgr","name":"mgr/rbd_support/b/trash_purge_schedule"}]: dispatch

9/26/22 12:29:49 PM
[INF]
from='mgr.54140 10.1.250.244:0/556836194' entity='mgr.b' cmd=[{"prefix":"config rm","who":"mgr","name":"mgr/rbd_support/b/trash_purge_schedule"}]: dispatch

9/26/22 12:29:49 PM
[INF]
from='mgr.54140 ' entity='mgr.b' cmd=[{"prefix":"config rm","who":"mgr","name":"mgr/rbd_support/b/mirror_snapshot_schedule"}]: dispatch

9/26/22 12:29:49 PM
[INF]
from='mgr.54140 10.1.250.244:0/556836194' entity='mgr.b' cmd=[{"prefix":"config rm","who":"mgr","name":"mgr/rbd_support/b/mirror_snapshot_schedule"}]: dispatch
```

I have also tried setting the ROOK_LOG_LEVEL to WARNING in the operator.yaml and also created a rook-config-override.yaml to override the configmap but it does not affect the audit log.
```
            apiVersion: v1
            kind: ConfigMap
            metadata:
              name: rook-config-override
              namespace: rook-ceph
            data:
              config: |
                [global]
                mon cluster log file level = error
                mon cluster log to syslog level = error
                [mgr]
                mon cluster log file level = error
                mon cluster log to syslog level = error
                [mon]
                mon cluster log file level = error
                mon cluster log to syslog level = error
                [osd]
                mon cluster log file level = error
                mon cluster log to syslog level = error
                [mds]
                mon cluster log file level = error
                mon cluster log to syslog level = error
```

I am actually pushing logs to Elasticsearch and Kibana is reporting excessive logs coming from rook-ceph.

I hope to get some advise from you guys.

Thank you & Best regards,
Serena

**Comment by travisn** [comment]
Have you configured snapshot schedules? Or this is just a default install? It's certainly unexpected for those configs to be repeating again and again every minute. I can't find anywhere in the rook operator or csi driver that runs these commands, so I suspect it's within one of the mgr modules. Perhaps you can disable the mgr module that is causing this if you're not using mirroring.

**Reply by serenagrl** [reply]
> Have you configured snapshot schedules? Or this is just a default install? It's certainly unexpected for those configs to be repeating again and again every minute. I can't find anywhere in the rook operator or csi driver that runs these commands, so I suspect it's within one of the mgr modules. Perhaps you can disable the mgr module that is causing this if you're not using mirroring.

Hi! It is a default installation from running `kubectl create -f` on the `crd.yaml, common.yaml, operator.yaml` and `cluster.yaml` in the `rook/deploy/examples` folder. I am unable to pinpoint which mgr module that is causing those logs.

**Reply by travisn** [reply]
@idryomov Do you know how we might be able to disable the verbose snapshot schedule messages in the mgr log?

**Reply by idryomov** [reply]
This is the cluster log (the audit channel of the cluster log, to be precise), not the mgr log.

I don't think the Dashboard is getting the cluster log from a log file (or syslog, or journald, etc), so `mon cluster log file level` and `mon cluster log to syslog level` appearing to not work isn't surprising.  This is a question for @epuertat.

**Reply by idryomov** [reply]
> It's certainly unexpected for those configs to be repeating again and again every minute. I can't find anywhere in the rook operator or csi driver that runs these commands, so I suspect it's within one of the mgr modules.

It's coming from rbd_support mgr module and, the way it's currently implemented, this is expected -- see https://github.com/ceph/ceph/pull/37864.  It should be easy to clean up to avoid spamming the audit channel though. I went ahead and filed https://tracker.ceph.com/issues/57726 for that.

**Reply by travisn** [reply]
@idryomov sounds good, thanks for the follow up!

**Reply by idryomov** [reply]
Hrm, looking a bit deeper it looks like `mon cluster log file level` actually got broken in quincy.  In this case the log file is irrelevant so it doesn't matter, but this appears to be a regression.  I have filed https://tracker.ceph.com/issues/57727, @neha-ojha FYI.

**Reply by epuertat** [reply]
> This is the cluster log (the audit channel of the cluster log, to be precise), not the mgr log.
> 
> I don't think the Dashboard is getting the cluster log from a log file (or syslog, or journald, etc), so `mon cluster log file level` and `mon cluster log to syslog level` appearing to not work isn't surprising. This is a question for @epuertat.

Yes @idryomov, that's the cluster log. I was expecting that the ceph-mgr provided some high-level API for that, but the Dashboard is simply invoking a `ceph log last` command through the `mon`.
